File: rpl/Exercises/B - Division y Conquista/17 - Mas de la mitad de repeticiones/main.c
#include "main.h" // No borrar

bool _mas_de_mitad(int *arr, size_t inicio, size_t fin, size_t n, int *contador) {

    if (*contador > n / 2)
        return true;

    int elem = arr[inicio];

    while (inicio <= fin) {
        if (elem == arr[inicio])
            *contador++;
        inicio++;
    }

    size_t medio = (inicio + fin) / 2;

    return _mas_de_mitad(arr, inicio, medio, (medio - inicio) + 1, &contador) && _mas_de_mitad(arr, medio, fin, (fin - medio) + 1, &contador);
}

bool mas_de_mitad(int* arr, size_t n) {
    if (n == 1)
        return true;

    int *contador = 0;
    
    return _mas_de_mitad(arr, 0, n-1, n, &contador);
}

File: rpl/Exercises/B - Division y Conquista/17 - Mas de la mitad de repeticiones/main.py
l = [9, 9, 9, 9, 9, 9, 9, 9]

def mas_de_mitad(l, inicio, fin, len):
    contador = 0
    j = inicio
    elem = l[j]
    mitad = len // 2

    while j <= fin:
        if elem == l[j]:
            contador += 1
        j += 1
    
    if contador > mitad:
        return True
    elif (inicio != fin):
        return mas_de_mitad(l, inicio+1, len-1, len)
    else:
        return False

print(mas_de_mitad(l, 0, len(l)-1, len(l)))
